x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/IndexPattern.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.esql.plan;
+
+import org.elasticsearch.xpack.esql.core.tree.Source;
+
+import java.util.Objects;
+
+/**
+ * Contains an index pattern together with its {@link Source}. Can also be a comma-separated list, like {@code idx-*,remote:other-idx*}.
+ */
+public class IndexPattern {
+
+    private final Source source;
+    private final String indexPattern;
```

Comment:
I wonder if this should be called indexPattern(s) as it might list coma separated patterns?

Not sure if it is needed often, but https://github.com/elastic/elasticsearch/pull/120277 would benefit if it is possible to return list/array of separate patterns.

Reply:
We can add a helper method as part of https://github.com/elastic/elasticsearch/pull/120277! I'd like to not add logic in this PR, so it remains a straight refactoring.

Reply:
Also, yeah, `indexPatterns` (plural) would be even more precise - but then we'd have to align e.g. `EsRelation` as well. And it's also reasonable to consider `idx,logs-*` a single pattern which matches `idx` and anything starting with `logs-`.